File: platforms/android/stato/src/main/java/com/facebook/stato/nativeplugins/components/ToolbarSidebarSection.java
package com.facebook.stato.nativeplugins.components;

import com.facebook.stato.core.StatoArray;
import com.facebook.stato.core.StatoObject;

public class ToolbarSidebarSection implements SidebarSection {
  StatoArray.Builder items = new StatoArray.Builder();

  public ToolbarSidebarSection addLink(String label, String destination) {
    items.put(
        new StatoObject.Builder()
            .put("type", "link")
            .put("label", label)
            .put("destination", destination));
    return this;
  }

  @Override
  public StatoObject serialize() {
    return new StatoObject.Builder().put("type", "toolbar").put("items", items.build()).build();
  }
}

File: platforms/android/android-stato/src/main/kotlin/org/stato/nativeplugins/components/ToolbarSidebarSection.kt
package org.stato.nativeplugins.components

import org.stato.core.StatoArray
import org.stato.core.StatoObject

class ToolbarSidebarSection : SidebarSection {

  private var items = StatoArray.Builder()

  fun addLink(label: String, destination: String): ToolbarSidebarSection {
    items.put(
      StatoObject.Builder()
        .put("type", "link")
        .put("label", label)
        .put("destination", destination))
    return this
  }

    override fun serialize(): StatoObject {
    return StatoObject.Builder().put("type", "toolbar").put("items", items.build()).build()
  }
}
